src/azul/indexer/document.py

```diff
@@ -1528,13 +1528,10 @@ class Replica(Document[ReplicaCoordinates[E]]):
     """
 
     #: The type of replica, i.e., what sort of metadata document from the
-    #: underlying data repository we are storing a copy of. Conceptually related
-    #: to the entity type, but its value may be different from the entity type.
-    #: For example, AnVIL replicas use the name of the data table that contains
-    #: the entity, e.g. 'anvil_file', instead of just 'file'.
+    #: underlying data repository we are storing a copy of. In practice, this is
+    #: the same as `self.coordinates.entity.entity_type`, but this isn't
+    # necessarily the case.
```

Comment:
> the commit Fix incorrect canned table names reads like a smell

The context here (for the whole PR) is that currently we only index from BigQuery tables whose names start with `anvil_`, so we often omit that prefix in our code and prepend it where/when it's needed. But soon we will start indexing from tables whose names do not include that prefix, so we have to start using the full explicit table names everywhere.

> it sounds like something was wrong prior

Sort of. The table names in the canned file are missing the `anvil_` prefix, but they end up being created with the correct names when the test is run, because the `_full_table_name` method inserts the prefix. But because we need to stop relying on this implicit insertion, this will soon stop working.

> in which case fixing it should happen first.

What do you mean by "first"? As the first commit of this PR? In a separate PR that is merged before this one? I don't think either option is appropriate. I thought that I *had* addressed it "first" by moving these changes to a separate commit on this PR that precedes the main commit(s).

> But the changes seems more like a re-adaptation, or an extension of the previous commit.

The previous commit ("Refactor table-column lookup in AnVIL plugin") is a refactoring that makes the second commit ("Fix incorrect canned table names") simpler and cleaner by introducing the `_schema_columns` attribute.

> Feel free to correct any misconceptions I may have.

I hope that I have done so.

Reply:
>I don't think either option is appropriate.

Agreed. That stemmed from my unfamiliarity with table name differences between HCA and AnVIL.
I appreciate you clarifying.